src/main/java/io/stargate/sgv2/jsonapi/service/operation/filters/table/codecs/VectorCodecs.java

```diff
@@ -26,38 +29,40 @@ public static <JavaT, CqlT> JSONCodec<JavaT, CqlT> arrayToCQLFloatVectorCodec(
     // (unless we want to rely on DB validating dimension as part of write and catch failure)
     return (JSONCodec<JavaT, CqlT>)
         new JSONCodec<>(
-            FLOAT_LIST,
+            FLOAT_LIST_TYPE,
             vectorType,
-            (cqlType, value) -> toCQLFloatVector(vectorType, value),
+            (cqlType, value) -> listToCQLFloatVector(vectorType, value),
             // This codec only for to-cql case, not to-json, so we don't need this
             null);
   }
 
+  public static <JavaT, CqlT> JSONCodec<JavaT, CqlT> binaryToCQLFloatVectorCodec(
+      VectorType vectorType) {
+    return (JSONCodec<JavaT, CqlT>)
+        new JSONCodec<>(
+            EJSON_TYPE,
+            vectorType,
+            (cqlType, value) -> binaryToCQLFloatVector(vectorType, value),
+            null);
+  }
+
   public static <JavaT, CqlT> JSONCodec<JavaT, CqlT> toJSONFloatVectorCodec(VectorType vectorType) {
     return (JSONCodec<JavaT, CqlT>)
         new JSONCodec<>(
-            FLOAT_LIST,
+            FLOAT_LIST_TYPE,
             vectorType,
             // This codec only for to-json case, not to-cql, so we don't need this
             null,
             (objectMapper, cqlType, value) -> toJsonNode(objectMapper, (CqlVector<Number>) value));
   }
 
-  static CqlVector<Float> toCQLFloatVector(VectorType vectorType, Collection<?> listValue)
+  /** Method for actual conversion from JSON Number Array into CQL Float Vector. */
+  static CqlVector<Float> listToCQLFloatVector(VectorType vectorType, Collection<?> listValue)
       throws ToCQLCodecException {
     Collection<JsonLiteral<?>> vectorIn = (Collection<JsonLiteral<?>>) listValue;
-    final int expLen = vectorType.getDimensions();
-    if (expLen != vectorIn.size()) {
-      throw new ToCQLCodecException(
-          vectorIn,
-          vectorType,
-          "expected vector of length "
-              + expLen
-              + ", got one with "
-              + vectorIn.size()
-              + " elements");
-    }
-    List<Float> floats = new ArrayList<>(expLen);
+    validateVectorLength(vectorType, vectorIn, vectorIn.size());
+
+    List<Float> floats = new ArrayList<>(vectorIn.size());
```

Comment:
I thought Jonathan Ellis said there is a way to pass binary encoded vector as input, if so we should use it.

Reply:
I couldn't find a method to do that but I'll double-check now.

Reply:
No, unfortunately `CqlVector<T extends Number>` is internally represented as

```
private final List<T> list;
```

so I don't think there is a way around that.

Unless we could use type other than `CqlVector` to bind `vector` column from?